src/Concordium/Client/Types/Contract/Info.hs

```diff
@@ -73,8 +68,16 @@ data ContractInfo
 
 -- TODO: Use CS.ContractSchema to avoid having to lookup by contract name.
 data ContractState =
-    WithSchema !CS.ModuleSchema !AE.Value  -- ^ The module schema and the decoded contract state.
-  | JustBytes !ByteString                  -- ^ The binary-encoded contract state.
+    WithSchema
+    {
+      wsModuleSchema :: !CS.ModuleSchema, -- ^ The module schema
+      wsValue :: !(Maybe AE.Value), -- ^ The decoded contract state, if schema type found for the state.
```

Comment:
I'm not sure I understand the idea here. Why is this a maybe? Why does the other case not apply in this case.

If this has to be like it is then please provide a comment as to why. The current comment would indicate that if this is Nothing then the `JustBytes` case would apply.